nixos/modules/services/x11/clight.nix

```diff
@@ -0,0 +1,112 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.services.clight;
+
+  toConf = { isAttr ? false }@args: v:
+    if builtins.isFloat v then toString v
+    else if isList v then "[ " + concatMapStringsSep ", " (toConf {}) v + " ]"
+    else if isAttr && isString v then toString v
+    else generators.toPretty {} v;
+
+  clightConf = {
+    inherit (config.location) latitude longitude;
+    gamma_temp = with cfg.temperature; [ day night ];
+  } // cfg.settings;
+
+  clightConfFile = pkgs.writeText "clight.conf"
+    (concatStringsSep "\n" (mapAttrsToList
+      (name: value: "${toConf { isAttr = true; } name} = ${toConf {} value};")
+      (filterAttrs
+        (_: value: value != null)
+        clightConf)));
+in {
+  options.services.clight = {
+    enable = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Whether to enable clight or not.
+      '';
+    };
+
+    temperature = {
+      day = mkOption {
+        type = types.int;
+        default = 5500;
+        description = ''
+          Colour temperature to use during the day, between
+          <literal>1000</literal> and <literal>25000</literal> K.
+        '';
+      };
+      night = mkOption {
+        type = types.int;
+        default = 3700;
+        description = ''
+          Colour temperature to use at night, between
+          <literal>1000</literal> and <literal>25000</literal> K.
+        '';
+      };
+    };
+
+    settings = mkOption {
+      type = types.attrs;
```

Comment:
You need to set a fully specified type here, like `attrsOf (nullOr (either int ...`, otherwise merging and priorities won't work correctly.